sdk/core/Azure.Core/api/Azure.Core.netcoreapp2.1.cs

```diff
@@ -194,6 +194,8 @@ public partial class RequestContext
         public RequestContext() { }
         public System.Threading.CancellationToken CancellationToken { get { throw null; } set { } }
         public Azure.ErrorOptions ErrorOptions { get { throw null; } set { } }
+        public void AddClassifier(int statusCode, bool isError) { }
+        public void AddClassifier(int statusCode, System.Func<Azure.Core.HttpMessage, bool> classify) { }
```

Comment:
Hmm, I didn't realize we were okay with using Funcs as parameters to methods.

Reply:
I don't know if we are, but I think we need it to enable scenarios for storage.  We want to be able to access the ErrorCode in addition to the StatusCode, and storage holds ErrorCode in a header, whereas the rest of Azure has it in the response payload.

Reply:
Considering this is already an advanced scenario, would it also make sense to have this overload take in a `MessageClassifier`? Although it requires users to create a new type, the implementation for `TryClassify` would be the same. Some advantages would be that the "try-classify" semantics are more transparent, and we potentially avoid having to new up an instance internally.

Reply:
Addressed.